src/server/game/Globals/ObjectMgr.cpp

```diff
@@ -4015,7 +4015,7 @@ void ObjectMgr::LoadQuests()
 
     // Load `quest_objectives`
     //                                   0   1        2     3             4         5       6      7       8                  9
-    result = WorldDatabase.Query("SELECT ID, QuestID, Type, StorageIndex, ObjectID, Amount, Flags, Flags2, ProgressBarWeight, Description FROM quest_objectives ORDER BY StorageIndex ASC");
+    result = WorldDatabase.Query("SELECT ID, QuestID, Type, StorageIndex, ObjectID, Amount, Flags, Flags2, ProgressBarWeight, Description FROM quest_objectives ORDER BY Order ASC");
```

Comment:
Well i recommended a new full parse.
But ordering them by Order AND Storage would break it again.

It should be order them by Order OR Storage.
-> If ObjectiveCount >= 2  and all Objectives have order 0 then order by storageIndex